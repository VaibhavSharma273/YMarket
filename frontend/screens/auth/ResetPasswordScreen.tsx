import React, { useState } from 'react';
import { StyleSheet } from 'react-native';

import API from '../../api/ymarket_api';

import { Text, View } from '../../components/Themed';
import TextInput from '../../components/TextInput';
import BackButton from '../../components/BackButton';
import { TouchableOpacity } from 'react-native';

import { theme } from '../../assets/theme';
import { emailValidator } from '../../helpers/emailValidator';
import { passwordValidator } from '../../helpers/passwordValidator';

export default function ResetPasswordScreen({ navigation }: any) {
    const [email, setEmail] = useState({ value: '', error: '' });

    const sendResetPasswordEmail = async () => {
        const emailError = emailValidator(email.value);
        if (emailError) {
            setEmail({ ...email, error: emailError });
            return;
        }

        const email_val = email.value;

        const response = await API.post('api/users/password-reset/', { email: email_val })
            .then((response) => {
                console.log(response.data);
                navigation.navigate('LoginScreen'); // dummy screen -- your email has been sent -- add go back to login button
            })
            .catch((error) => {
                console.log(error);
                if (error.response) {
                    setEmail({ ...email, error: error.response.data[Object.keys(error.response.data)[0]] });
                }
            });

        navigation.navigate('LoginScreen');
    };

    return (
        <View style={styles.container}>
            <BackButton goBack={navigation.goBack} />
            {/* <View style={styles.separator} /> */}
            <Text style={styles.header}>Restore Password</Text>
            <TextInput
                label="Yale Email"
                returnKeyType="done"
                value={email.value}
                onChangeText={(text: any) => setEmail({ value: text, error: '' })}
                error={!!email.error}
                errorText={email.error}
                autoCapitalize="none"
                autoCompleteType="email"
                textContentType="emailAddress"
                keyboardType="email-address"
                description="You will receive an email with a password reset link."
            />

            <TouchableOpacity style={styles.button} onPress={sendResetPasswordEmail}>
                <Text style={styles.title}>Send Instructions</Text>
            </TouchableOpacity>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
    },
    title: {
        fontSize: 20,
        fontWeight: 'bold',
        color: 'white',
        fontFamily: 'Arial',
    },
    row: {
        flexDirection: 'row',
        marginTop: 4,
    },
    link: {
        fontWeight: 'bold',
        color: theme.colors.primary,
    },
    header: {
        fontSize: 21,
        color: '#0f4d92',
        fontWeight: 'bold',
        paddingVertical: 12,
    },
    button: {
        alignItems: 'center',
        backgroundColor: '#0f4d92',
        padding: 10,
        marginTop: 20,
    },
    separator: {
        marginVertical: 6,
        height: 3,
        width: '50%',
    },
    forgotPassword: {
        width: '100%',
        alignItems: 'flex-end',
        marginBottom: 24,
    },
    forgot: {
        fontSize: 13,
        color: 'gray',
    },
});
