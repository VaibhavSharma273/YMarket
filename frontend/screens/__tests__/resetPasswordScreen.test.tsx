import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import ResetPasswordScreen from '../auth/ResetPasswordScreen';
import { createTestProps } from '../../jest.setup';
import renderer from 'react-test-renderer';

describe('Reset Password Screen component', () => {
    let component: ShallowWrapper;
    let props: any;
    beforeEach(() => {
        props = createTestProps({});
        component = shallow(<ResetPasswordScreen {...props} />);
    });
    it(`renders correctly`, () => {
        expect(component.length).toBe(1);
    });
    it('matches the snapshot', () => {
        const tree = renderer.create(<ResetPasswordScreen {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
