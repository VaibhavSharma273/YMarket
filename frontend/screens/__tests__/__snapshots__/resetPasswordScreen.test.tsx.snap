// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Screen component matches the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 4,
        "opacity": 1,
        "position": "absolute",
        "top": 30,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 24,
          "width": 24,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "#0f4d92",
          "fontSize": 21,
          "fontWeight": "bold",
          "paddingVertical": 12,
        },
      ]
    }
  >
    Restore Password
  </Text>
  <View
    style={
      Object {
        "marginVertical": 1,
        "width": "80%",
      }
    }
  >
    <View
      style={Object {}}
    >
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "rgba(0, 0, 0, 0.54)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
          ]
        }
        testID="text-input-outline"
      />
      <View>
        <View
          style={
            Array [
              Object {
                "paddingBottom": 0,
              },
              Object {
                "minHeight": 64,
                "paddingTop": 8,
              },
            ]
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "zIndex": 3,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "bottom": 4,
                  "left": 10,
                  "maxHeight": 2,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 12,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "maxWidth": -28,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Yale Email
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#0f4d92",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Yale Email
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Yale Email
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCompleteType="email"
            editable={true}
            keyboardType="email-address"
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            returnKeyType="done"
            selectionColor="#0f4d92"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                Array [
                  Object {},
                ],
              ]
            }
            testID="text-input-outlined"
            textContentType="emailAddress"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 13,
          "paddingTop": 0,
        }
      }
    >
      You will receive an email with a password reset link.
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0f4d92",
        "marginTop": 20,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "white",
            "fontFamily": "Arial",
            "fontSize": 20,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Send Instructions
    </Text>
  </View>
</View>
`;
